Carthage update failing

I use DKImagePickerController on my project using carthage.

When I run carthage update, DKPhotoGallery seems to fail while building.

Looking at the logs I found this:

`Command CompileSwift failed with a nonzero exit code`

And between the lines, the following that may be causing the build to fail (it is not the full log):

```
DKPhotoGallery.swift:445:11: warning: 'public' modifier is redundant for instance method declared in a public extension
    @objc public func present(photoGallery gallery: DKPhotoGallery, completion: (() -> Swift.Void)? = nil) {
          ^~~~~~~
          
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoImagePreviewVC.swift:129:44: error: cannot call value of non-function type 'SDImageCache'
            var image = SDImageCache.shared().imageFromMemoryCache(forKey: key)
                        ~~~~~~~~~~~~~~~~~~~^~
                                           
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoImagePreviewVC.swift:134:44: error: cannot call value of non-function type 'SDImageCache'
                        SDImageCache.shared().store(nil, imageData: data, forKey: key, toDisk: false, completion: nil)
                        ~~~~~~~~~~~~~~~~~~~^~
                                           
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoImagePreviewVC.swift:173:28: error: cannot call value of non-function type 'SDImageCache'
        SDImageCache.shared().queryCacheOperation(forKey: key, options: SDImageCacheOptions(rawValue: 1 << 2).union(.queryDataWhenInMemory)) { (image, data, cacheType) in
        ~~~~~~~~~~~~~~~~~~~^~
                           
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoImagePreviewVC.swift:241:36: error: cannot call value of non-function type 'SDImageCache'
                SDImageCache.shared().queryCacheOperation(forKey: originalURL.absoluteString, options: SDImageCacheOptions(rawValue: 1 << 2)) { (image, data, _) in
                ~~~~~~~~~~~~~~~~~~~^~
                                   
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoImagePreviewVC.swift:268:36: error: cannot call value of non-function type 'SDImageCache'
                SDImageCache.shared().queryCacheOperation(forKey: originalURL.absoluteString) { [weak self] (image, data, _) in
                ~~~~~~~~~~~~~~~~~~~^~
                                   
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoImageView.swift:17:21: error: initializer does not override a designated initializer from its superclass
    public override init(image: UIImage? = nil, highlightedImage: UIImage? = nil) {
           ~~~~~~~~ ^
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoImageView.swift:15:30: error: use of undeclared type 'FLAnimatedImageView'
open class DKPhotoImageView: FLAnimatedImageView {
                             ^~~~~~~~~~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoImageView.swift:18:9: error: 'super' members cannot be referenced in a root class
        super.init(image: image, highlightedImage: highlightedImage)
        ^
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoImageView.swift:20:9: error: value of type 'DKPhotoImageView' has no member 'contentMode'
        self.contentMode = .scaleAspectFit
        ^~~~ ~~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:24:50: warning: cast from 'UIView?' to unrelated type 'DKPhotoImageView' always fails
        guard let contentView = self.contentView as? DKPhotoImageView else { return nil }
                                ~~~~~~~~~~~~~~~~ ^   ~~~~~~~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:26:33: error: value of type 'DKPhotoImageView' has no member 'image'
        guard let targetImage = contentView.image ?? contentView.animatedImage?.posterImage else {
                                ^~~~~~~~~~~ ~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:68:50: warning: cast from 'UIView?' to unrelated type 'DKPhotoImageView' always fails
        guard let contentView = self.contentView as? DKPhotoImageView else { return }
                                ~~~~~~~~~~~~~~~~ ^   ~~~~~~~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:74:44: error: value of type 'DKPhotoImageView' has no member 'animatedImage'
                    if let animatedImage = contentView.animatedImage {
                                           ^~~~~~~~~~~ ~~~~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:82:43: error: value of type 'DKPhotoImageView' has no member 'image'
                    } else if let image = contentView.image {
                                          ^~~~~~~~~~~ ~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:112:16: error: cannot convert return expression of type 'DKPhotoImageView' to return type 'UIView'
        return contentView
               ^~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:116:50: warning: cast from 'UIView?' to unrelated type 'DKPhotoImageView' always fails
        guard let contentView = self.contentView as? DKPhotoImageView else { return }
                                ~~~~~~~~~~~~~~~~ ^   ~~~~~~~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:121:45: error: use of unresolved identifier 'FLAnimatedImage'; did you mean 'SDAnimatedImage'?
                contentView.animatedImage = FLAnimatedImage(gifData: data)
                                            ^~~~~~~~~~~~~~~
                                            SDAnimatedImage
SDWebImage.SDAnimatedImage:1:12: note: 'SDAnimatedImage' declared here
open class SDAnimatedImage : UIImage, SDAnimatedImageProtocol {
           ^
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:123:17: error: value of type 'DKPhotoImageView' has no member 'image'
                contentView.image = UIImage(data: data)
                ^~~~~~~~~~~ ~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:126:13: error: value of type 'DKPhotoImageView' has no member 'image'
            contentView.image = image
            ^~~~~~~~~~~ ~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:133:47: warning: cast from 'UIView?' to unrelated type 'DKPhotoImageView' always fails
        if let contentView = self.contentView as? DKPhotoImageView {
                             ~~~~~~~~~~~~~~~~ ^   ~~~~~~~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:134:28: error: value of type 'DKPhotoImageView' has no member 'image'
            if let image = contentView.image {
                           ^~~~~~~~~~~ ~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:136:23: error: value of type 'DKPhotoImageView' has no member 'animatedImage'
            } else if contentView.animatedImage != nil {
                      ^~~~~~~~~~~ ~~~~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:137:24: error: value of type 'DKPhotoImageView' has no member 'currentFrame'
                return contentView.currentFrame
                       ^~~~~~~~~~~ ~~~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:149:50: warning: cast from 'UIView?' to unrelated type 'DKPhotoImageView' always fails
        guard let contentView = self.contentView as? DKPhotoImageView else { return }
                                ~~~~~~~~~~~~~~~~ ^   ~~~~~~~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:151:9: error: value of type 'DKPhotoImageView' has no member 'image'
        contentView.image = DKPhotoGalleryResource.downloadFailedImage()
        ^~~~~~~~~~~ ~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:152:9: error: value of type 'DKPhotoImageView' has no member 'contentMode'
        contentView.contentMode = .center
        ^~~~~~~~~~~ ~~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:160:50: warning: cast from 'UIView?' to unrelated type 'DKPhotoImageView' always fails
        guard let contentView = self.contentView as? DKPhotoImageView else { return CGSize.zero }
                                ~~~~~~~~~~~~~~~~ ^   ~~~~~~~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:162:24: error: value of type 'DKPhotoImageView' has no member 'image'
        if let image = contentView.image {
                       ^~~~~~~~~~~ ~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoBaseImagePreviewVC.swift:164:39: error: value of type 'DKPhotoImageView' has no member 'animatedImage'
        } else if let animatedImage = contentView.animatedImage {
                                      ^~~~~~~~~~~ ~~~~~~~~~~~~~
/Users/myuser/Desktop/test/Carthage/Checkouts/DKPhotoGallery/DKPhotoGallery/Preview/ImagePreview/DKPhotoImageDownloader.swift:36:9: error: value of type 'SDWebImageDownloader' has no member 'executionOrder'
        downloader.executionOrder = .lifoExecutionOrder
```
Update: Using Xcode 10.2 and Carthage 0.33

--- a/Cartfile
+++ b/Cartfile
@@ -1,1 +1,1 @@
-github "rs/SDWebImage"
+github "SDWebImage/SDWebImageFLPlugin"

--- a/Cartfile.resolved
+++ b/Cartfile.resolved
@@ -1,1 +1,3 @@
-github "rs/SDWebImage" "4.4.2"
+github "Flipboard/FLAnimatedImage" "1.0.14"
+github "SDWebImage/SDWebImage" "5.0.2"
+github "SDWebImage/SDWebImageFLPlugin" "0.3.0"

--- a/DKPhotoGallery.podspec
+++ b/DKPhotoGallery.podspec
@@ -12,7 +12,8 @@ Pod::Spec.new do |s|
     s.frameworks    = 'Foundation', 'UIKit', 'Photos', 'WebKit', 'AVFoundation', 'AVKit', 'AssetsLibrary', 'MobileCoreServices'
     s.requires_arc  = true
     s.swift_version = '4.2'
-    s.dependency 'SDWebImage/GIF'
+    s.dependency 'SDWebImage'
+    s.dependency 'SDWebImageFLPlugin'
 
     s.subspec 'Core' do |core|
         core.dependency 'DKPhotoGallery/Model'

--- a/DKPhotoGallery.xcodeproj/project.pbxproj
+++ b/DKPhotoGallery.xcodeproj/project.pbxproj
@@ -7,6 +7,8 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		8B40015C2274F13200061553 /* SDWebImageFLPlugin.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 8B40015A2274F13200061553 /* SDWebImageFLPlugin.framework */; };
+		8B40015D2274F13200061553 /* FLAnimatedImage.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 8B40015B2274F13200061553 /* FLAnimatedImage.framework */; };
 		9C565E32216B43DA00A16B22 /* Base.lproj in Resources */ = {isa = PBXBuildFile; fileRef = 9C565E2E216B43DA00A16B22 /* Base.lproj */; };
 		9C565E33216B43DA00A16B22 /* Images.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 9C565E2F216B43DA00A16B22 /* Images.xcassets */; };
 		9C565E34216B43DA00A16B22 /* zh-Hans.lproj in Resources */ = {isa = PBXBuildFile; fileRef = 9C565E30216B43DA00A16B22 /* zh-Hans.lproj */; };
@@ -59,6 +61,8 @@
 /* End PBXContainerItemProxy section */
 
 /* Begin PBXFileReference section */
+		8B40015A2274F13200061553 /* SDWebImageFLPlugin.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = SDWebImageFLPlugin.framework; path = Carthage/Build/iOS/SDWebImageFLPlugin.framework; sourceTree = "<group>"; };
+		8B40015B2274F13200061553 /* FLAnimatedImage.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = FLAnimatedImage.framework; path = Carthage/Build/iOS/FLAnimatedImage.framework; sourceTree = "<group>"; };
 		9C08E2A5215B6280003BA0DA /* module.modulemap */ = {isa = PBXFileReference; lastKnownFileType = "sourcecode.module-map"; path = module.modulemap; sourceTree = "<group>"; };
 		9C17CC0E215B39C1001A6BCA /* Info.plist */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
 		9C565E26216B42AE00A16B22 /* DKPhotoGallery.bundle */ = {isa = PBXFileReference; explicitFileType = wrapper.cfbundle; includeInIndex = 0; path = DKPhotoGallery.bundle; sourceTree = BUILT_PRODUCTS_DIR; };
@@ -109,6 +113,8 @@
 			isa = PBXFrameworksBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
+				8B40015C2274F13200061553 /* SDWebImageFLPlugin.framework in Frameworks */,
+				8B40015D2274F13200061553 /* FLAnimatedImage.framework in Frameworks */,
 				9C90C060215E371300314C91 /* SDWebImage.framework in Frameworks */,
 				9CF878DF215B2F91003F2667 /* MobileCoreServices.framework in Frameworks */,
 				9CF878DD215B2F85003F2667 /* WebKit.framework in Frameworks */,
@@ -159,6 +165,8 @@
 		9CDF249D215A46FD00DF2C79 /* Frameworks */ = {
 			isa = PBXGroup;
 			children = (
+				8B40015B2274F13200061553 /* FLAnimatedImage.framework */,
+				8B40015A2274F13200061553 /* SDWebImageFLPlugin.framework */,
 				9C90C05E215E371300314C91 /* SDWebImage.framework */,
 				9CF878DE215B2F91003F2667 /* MobileCoreServices.framework */,
 				9CF878DC215B2F85003F2667 /* WebKit.framework */,

--- a/DKPhotoGallery/Preview/ImagePreview/DKPhotoImageDownloader.swift
+++ b/DKPhotoGallery/Preview/ImagePreview/DKPhotoImageDownloader.swift
@@ -32,8 +32,9 @@ class DKPhotoImageWebDownloader: DKPhotoImageDownloader {
     }
     
     var _downloader: SDWebImageDownloader = {
-        let downloader = SDWebImageDownloader()
-        downloader.executionOrder = .lifoExecutionOrder
+        let config = SDWebImageDownloaderConfig()
+        config.executionOrder = .lifoExecutionOrder
+        let downloader = SDWebImageDownloader.init(config: config)
         
         return downloader
     }()

--- a/DKPhotoGallery/Preview/ImagePreview/DKPhotoImagePreviewVC.swift
+++ b/DKPhotoGallery/Preview/ImagePreview/DKPhotoImagePreviewVC.swift
@@ -126,18 +126,18 @@ class DKPhotoImagePreviewVC: DKPhotoBaseImagePreviewVC {
         DKPhotoImagePreviewVC.ioQueue.async {
             let key = URL.absoluteString
             
-            var image = SDImageCache.shared().imageFromMemoryCache(forKey: key)
+            var image = SDImageCache.shared.imageFromMemoryCache(forKey: key)
             if image == nil {
                 do {
                     let data = try Data(contentsOf: URL)
                     if NSData.sd_imageFormat(forImageData: data) == .GIF {
-                        SDImageCache.shared().store(nil, imageData: data, forKey: key, toDisk: false, completion: nil)
+                        SDImageCache.shared.store(nil, imageData: data, forKey: key, toDisk: false, completion: nil)
                         
                         completeBlock(data, nil)
                     } else if let compressedImage = UIImage.sd_image(with: data) {
                         image = compressedImage.decompress()
                         
-                        SDImageCache.shared().store(image, forKey: key, toDisk: false, completion: nil)
+                        SDImageCache.shared.store(image, forKey: key, toDisk: false, completion: nil)
                         
                         completeBlock(image, nil)
                     } else {
@@ -170,7 +170,7 @@ class DKPhotoImagePreviewVC: DKPhotoBaseImagePreviewVC {
         }
         
         // SDWebImage 4.4.0: SDImageCacheScaleDownLargeImages   1 << 2
-        SDImageCache.shared().queryCacheOperation(forKey: key, options: SDImageCacheOptions(rawValue: 1 << 2).union(.queryDataWhenInMemory)) { (image, data, cacheType) in
+        SDImageCache.shared.queryCacheOperation(forKey: key, options: SDImageCacheOptions(rawValue: 1 << 2).union(.queryMemoryData)) { (image, data, cacheType) in
             if image != nil || data != nil {
                 if NSData.sd_imageFormat(forImageData: data) == .GIF {
                     completeBlock(data ?? image, nil)
@@ -182,7 +182,7 @@ class DKPhotoImagePreviewVC: DKPhotoBaseImagePreviewVC {
                     progressBlock(progress)
                 }, completeBlock: { (image, data, error) in
                     if error == nil {
-                        SDImageCache.shared().store(image, imageData: data, forKey: key, toDisk: true, completion: nil)
+                        SDImageCache.shared.store(image, imageData: data, forKey: key, toDisk: true, completion: nil)
                         
                         completeBlock(data ?? image, nil)
                     } else {
@@ -238,7 +238,7 @@ class DKPhotoImagePreviewVC: DKPhotoBaseImagePreviewVC {
         if let downloadURL = self.downloadURL {
             if let extraInfo = self.item.extraInfo, let originalURL = extraInfo[DKPhotoGalleryItemExtraInfoKeyRemoteImageOriginalURL] as? URL {
                 // SDWebImage 4.4.0: SDImageCacheScaleDownLargeImages   1 << 2
-                SDImageCache.shared().queryCacheOperation(forKey: originalURL.absoluteString, options: SDImageCacheOptions(rawValue: 1 << 2)) { (image, data, _) in
+                SDImageCache.shared.queryCacheOperation(forKey: originalURL.absoluteString, options: SDImageCacheOptions(rawValue: 1 << 2)) { (image, data, _) in
                     if image != nil || data != nil {
                         self.downloadURL = originalURL
                     }
@@ -265,7 +265,7 @@ class DKPhotoImagePreviewVC: DKPhotoBaseImagePreviewVC {
                 self.downloadOriginalImageButton.isHidden = true
             } else {
                 let reuseIdentifier = self.reuseIdentifier
-                SDImageCache.shared().queryCacheOperation(forKey: originalURL.absoluteString) { [weak self] (image, data, _) in
+                SDImageCache.shared.queryCacheOperation(forKey: originalURL.absoluteString) { [weak self] (image, data, _) in
                     guard reuseIdentifier == self?.reuseIdentifier else { return }
                     
                     if image != nil || data != nil {

--- a/DKPhotoGalleryDemo/DKPhotoGalleryDemo.xcodeproj/project.pbxproj
+++ b/DKPhotoGalleryDemo/DKPhotoGalleryDemo.xcodeproj/project.pbxproj
@@ -257,12 +257,14 @@
 				"${BUILT_PRODUCTS_DIR}/DKPhotoGallery/DKPhotoGallery.framework",
 				"${BUILT_PRODUCTS_DIR}/FLAnimatedImage/FLAnimatedImage.framework",
 				"${BUILT_PRODUCTS_DIR}/SDWebImage/SDWebImage.framework",
+				"${BUILT_PRODUCTS_DIR}/SDWebImageFLPlugin/SDWebImageFLPlugin.framework",
 			);
 			name = "[CP] Embed Pods Frameworks";
 			outputPaths = (
 				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/DKPhotoGallery.framework",
 				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/FLAnimatedImage.framework",
 				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/SDWebImage.framework",
+				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/SDWebImageFLPlugin.framework",
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;

